### PR TITLE
Typecast token to a buffer representation of a hex based string

### PR DESCRIPTION
**Pull request by retrohacker**

### CODE REVIEW

File: bin/actions/uploader.js [hunk 1]
```diff
@@ -212,8 +212,10 @@ Uploader.prototype._makeTempDir = function(filepath, callback) {
     if (token.encryptionKey) {
       // generate file key based on public encryptionKey
       var fileKey = storj.DeterministicKeyIv.getDeterministicKey(
-        token.encryptionKey, fileId);
-      secret = new storj.DeterministicKeyIv(fileKey, fileId);
+        Buffer.from(String(token.encryptionKey), 'hex'),
+        Buffer.from(String(fileId), 'hex'));
+      secret = new storj.DeterministicKeyIv(fileKey,
+        Buffer.from(String(fileId), 'hex'));
```
Comment (braydonf) [inline bin/actions/uploader.js]: This one I don't think needs to be a buffer.

**Comment by braydonf** [comment]
May not be necessary with https://github.com/Storj/core/pull/590/files

File: bin/actions/uploader.js (hunk 1 above)
Comment (retrohacker) [inline bin/actions/uploader.js]: https://github.com/Storj/core/blob/5586c5b50d36e7be5b3e73646552413ae9b39ffb/lib/crypto-tools/deterministic-key-iv.js#L29-L30

Comment (retrohacker) [inline bin/actions/uploader.js]: https://github.com/Storj/core/blob/5586c5b50d36e7be5b3e73646552413ae9b39ffb/lib/crypto-tools/deterministic-key-iv.js#L18-L19

Would be weird if 1/2 the object were a string and the other 1/2 a Buffer, no?

Comment (braydonf) [inline bin/actions/uploader.js]: `_pass` on the object is a string, see: https://github.com/Storj/core/blob/master/lib/crypto-tools/deterministic-key-iv.js#L49

Comment (braydonf) [inline bin/actions/uploader.js]: FYI: `DeterministicKeyIv.getDeterministicKey`, now also accepts hex strings: https://github.com/Storj/core/pull/590/files

**Comment by retrohacker** [comment]
Tested with 590. LGTM.

**Pull request closed without merging**